src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -30,4 +30,8 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
       case _ => Failure(new UnsupportedOperationException("Can only TSV serialize an Array[Primitive]"))
     }
   }
+
+  override def collect[T <: WdlValue](filterFn: PartialFunction[WdlValue, T]): Seq[T] = {
+    value flatMap { _.collect(filterFn) }
+  }
```

Comment:
The name `collect` bothers me a little, because this method doesn't return a `WdlArray`, but a `Seq` instead. The scala collections [`collect`](http://www.scala-lang.org/api/2.11.7/index.html#scala.collection.GenTraversableLike@collect[B](pf:PartialFunction[A,B]):scala.collection.GenTraversable[B]) round trips the container type after filtering. If it's not going to return the original type, maybe call it `collectSeq`, but I'm terrible at naming.

This isn't a blocker to merging, just something to think about. The same goes for the other collect-to-Seq methods too.